server/server.go

```diff
@@ -875,6 +881,29 @@ func (s *server) getClusterInfo() *rpcpb.ClusterInfo {
 	return info
 }
 
+func (s *server) handleUnexpectedNodeStop() {
+	unexpectedNodeStopCh, err := s.network.nw.GetUnexpectedNodeStopChannel()
+	if err != nil {
+		panic(err)
+	}
+	for {
+		nodeName := <-unexpectedNodeStopCh
+		zap.L().Info("received unexpected node stop message", zap.String("node-name", nodeName))
+		s.mu.Lock()
+		if s.clusterInfo == nil {
+			s.mu.Unlock()
+			return
+		}
+		_, ok := s.network.nodeInfos[nodeName]
+		if ok {
```

Comment:
Is `ok` ever `false`? I didn't look into it but my guess is it's not supposed to ever be `false`, and if that's the case we should log a warning that something that wasn't supposed to happen, did

Reply:
it is not supposed to happen but this are different objects maintained with separate logic so it is a fail safe